src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidations.java

```diff
@@ -198,12 +199,40 @@ private CallbackValidations() {
             .map(c -> (String) c);
     }
 
+    private static Optional<String> getAwaitingPaymentDcnProcessing(CaseDetails theCase) {
+        return Optional.ofNullable(theCase)
+            .map(CaseDetails::getData)
+            .map(data -> data.get("awaitingPaymentDCNProcessing"))
+            .map(c -> (String) c);
+    }
+
     static boolean hasOcr(CaseDetails theCase) {
         return getOcrData(theCase)
             .map(CollectionUtils::isNotEmpty)
             .orElse(false);
     }
 
+    static Validation<String, Void> validatePayments(
+        CaseDetails theCase,
+        Classification classification,
+        ServiceConfigItem config
+    ) {
+        Optional<String> awaitingPaymentsOptional = getAwaitingPaymentDcnProcessing(theCase);
+
+        if (!awaitingPaymentsOptional.isPresent()
+            || (awaitingPaymentsOptional.get().equals("No") // no payments or no payments pending for completion
+                || (awaitingPaymentsOptional.get().equals("Yes") // payments processing pending
+                    && config.getAllowAttachingToCaseBeforePaymentsAreProcessedForClassifications()
+                        .contains(classification)))
+        ) {
+            return valid(null);
+        } else {
+            return invalid(
+                "The 'attach to case' event is not supported for the Exception Record with pending payments"
+            );
+        }
```

Comment:
that main if statement.. maybe we could reverse it and approach in functional way?

```java
return getAwaitingPaymentDcnProcessing(theCase)
    .filter(flag -> !flag.equals("No"))
    .filter(flag -> !(flag.equals("Yes") && config.contains(classification))
    .map(voidFlag -> invalid(...))
    .orElse(valid(null));
```

p.s. comments are great for clarity ^^ ;)

Reply:
thanks, but the updated  IF statement is much simpler